needle.pm

```diff
@@ -354,24 +337,18 @@ sub tags {
     my $goods     = $tags{$first_tag};
 
     # go out early if there is nothing to do
-    if (!$goods || !@wanted) {
-        return $goods || [];
-    }
+    return $goods || [] unless $goods && @wanted;
     my @results;
 
     # now check that it contains all the other tags too
   NEEDLE: for my $n (@$goods) {
-        for my $t (@wanted) {
-            next NEEDLE if (!$n->has_tag($t));
-        }
+        next NEEDLE if (!$n->has_tag($_)) for (@wanted);
```

Comment:
I'm surprised that Perl can parse this. Are you sure this works? If it does I'd still remove the braces.

Reply:
who ever stated that this would work? :D https://travis-ci.org/github/os-autoinst/os-autoinst/builds/745513140 tells "no"

Reply:
At the time I was writing the CI was still pending but I assumed you would have at least run the needles test locally.

Reply:
no, sorry :blush: